### PR TITLE
varsutil: remove dependency on 'types' for 'variable' package

### PR DESCRIPTION
**Pull request by coocood**
'variable' package will be used in 'types', remove dependency on 'types' by
moving functions to a new 'varsutil' package.

Also extract 'SetExec' from 'SimpleExec'

### CODE REVIEW

**Comment by coocood** [comment]
@zimulala @hanfei1991 PTAL

**Comment by zimulala** [comment]
LGTM

File: sessionctx/varsutil/varsutil_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// limitations under the License.
+
+package varsutil
+
+import (
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/testleak"
+	"github.com/pingcap/tidb/util/types"
+	"testing"
```
Comment (hanfei1991) [inline sessionctx/varsutil/varsutil_test.go]: adjust package order.

**Comment by hanfei1991** [comment]
restLGTM

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**